### ISSUE TITLE
Docker compose setup fails for Umbraco on windows

### ISSUE THREAD

**Issue by emimontesdeoca** [umbraco/Umbraco-CMS#17648]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

15.0.0

### Bug summary

**Issue Description:**
While trying to set up a new Umbraco project with Docker using Docker Compose on a Windows environment, the database container fails to start due to script execution errors.

![image](https://github.com/user-attachments/assets/fdac24ac-f2c6-4b78-b06a-bf89875105ee)

**System Information:**
- **OS:** Windows
- **Docker Version:** Docker Desktop for Windows (with WSL 2 backend)
- **Umbraco Version:** (use specific version if known, otherwise write the latest available version as of description date)

### Specifics

Using the official documentation: https://docs.umbraco.com/umbraco-cms/fundamentals/setup/install/running-umbraco-on-docker-locally

```
PS C:\Development\Personal\MyDockerProject> docker version
Client:
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        ce12230
 Built:             Fri Sep 20 11:42:27 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.35.1 (173168)
 Engine:
  Version:          27.3.1
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.22.7
  Git commit:       41ca978
  Built:            Fri Sep 20 11:41:11 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.21
  GitCommit:        472731909fa34bd7bc9c087e4c27943f9835f111
 runc:
  Version:          1.1.13
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
PS C:\Development\Personal\MyDockerProject>
```

### Steps to reproduce

1. Create a new directory for the Docker project:
   ```powershell
   mkdir MyDockerProject
   ```
2. Change to the newly created directory:
   ```powershell
   cd MyDockerProject
   ```
3. Create a new Umbraco project with Docker support:
   ```powershell
   dotnet new umbraco -n MyDockerProject --add-docker
   ```
4. Add Docker Compose support for the Umbraco project:
   ```powershell
   dotnet new umbraco-compose -P "MyDockerProject"
   ```
5. Start the Docker Compose setup:
   ```powershell
   docker compose up
   ```

### Expected result / actual result

**Expected Behavior:**
The Docker containers for the Umbraco project and its associated database should start successfully without any errors.

**Actual Behavior:**
The database container fails to start, outputting the following errors:
```
: invalid optiondatabase  | startup.sh: line 2: set: -
mydockerproject_database  | set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
mydockerproject_database  | startup.sh: line 3: $'\r': command not found
mydockerproject_database  | startup.sh: line 9: syntax error near unexpected token `$'{\r''
'ydockerproject_database  | startup.sh: line 9: `    function initialize_app_database() {
mydockerproject_database exited with code 2
```

---
_This item has been added to our backlog AB#46478_

**Comment by nikolajlauridsen** [comment]
Hey @emimontesdeoca thanks for reporting, I can confirm that this is an issue 😄 

The issue stems from Windows being used to build the release, resolving the wrong line endings. 

Additionally I've found an issue that the wrong dotnet version is used in the docker file. 

I'll get a fix made, but for now you can work around this by:

1. Open startup.sh in an editor like VSCode, and change the line endings from `CRLF` to `LF
2. Open op the `Dockerfile` and change both instances of `FROM mcr.microsoft.com/dotnet/sdk:8.0 AS` to `FROM mcr.microsoft.com/dotnet/sdk:9.0 AS`

**Comment by rasmusberntsen-merkle** [comment]
> Hey @emimontesdeoca thanks for reporting, I can confirm that this is an issue 😄
> 
> The issue stems from Windows being used to build the release, resolving the wrong line endings.
> 
> Additionally I've found an issue that the wrong dotnet version is used in the docker file.
> 
> I'll get a fix made, but for now you can work around this by:
> 
> 1. Open startup.sh in an editor like VSCode, and change the line endings from `CRLF` to `LF
> 2. Open op the `Dockerfile` and change both instances of `FROM mcr.microsoft.com/dotnet/sdk:8.0 AS` to `FROM mcr.microsoft.com/dotnet/sdk:9.0 AS`

Just an FYI: I tested this fix locally and it's now working for me. I had to fix the line endings before doing anything, not enough just to update the startup.sh file after it has thrown the error. Maybe obvious for other people, but that took me some time to figure out. Haha. :)

**Comment by nikolajlauridsen** [comment]
I've made a fix for this here #17722, just need to get it merged. The problem was that our build server now runs on windows, so git be default checked out the files as `CRLF`, this has been fixed so all `.sh` files are now checked out as `LF`


In regards to having to do the fix before the error @rasmusberntsen-merkle it might be because docker only builds the first time, and then reuses the same image, you can always force a new build using `docker compose up --build`  😄

**Comment by nikolajlauridsen** [comment]
#17722 has been merged, and will be included in 15.1, so I'll go ahead and close this one, thanks for reporting 😄
